Composer/packages/server/src/controllers/eject.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import path from 'path';
+
+import { pluginLoader, PluginLoader } from '../services/pluginLoader';
+import { BotProjectService } from '../services/project';
+import { copyDir } from '../utility/storage';
+import { LocalDiskStorage } from '../models/storage/localDiskStorage';
+
+export const EjectController = {
+  getTemplates: async (req, res) => {
+    res.json(pluginLoader.extensions.runtimeTemplates);
+  },
+  eject: async (req, res) => {
+    const user = await PluginLoader.getUserFromRequest(req);
+    const projectId = req.params.projectId;
+    const currentProject = await BotProjectService.getProjectById(projectId, user);
+
+    const source = pluginLoader.extensions.runtimeTemplates.filter(i => i.key === req.params.template);
+    if (source.length) {
+      const template = source[0];
+      const runtimePath = path.join(currentProject.dir, 'runtime');
+      if (!(await currentProject.fileStorage.exists(runtimePath))) {
+        const fileStorage = new LocalDiskStorage();
```

Comment:
this will now do `await template.eject(project, new LocalFileStorage())`

longer term, we need to figure out a good way to expose some core interfaces (composerpluginregistration, ifilestorage, useridentity, a few others...) to the plugins otherwise they have to be recreated inside the plugin.